### ISSUE TITLE
devon ide update not working on windows

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#212]
For `devon ide update` the `devon-ide` performs an update by moving existing stuff into backup and then putting the latest downloaded and extracted stuff into the original place.
However, when you are running a script then windows holds a file lock on that script. Hence, windows fails to move away the script file that is currently performing the update process:
```
Creating backup by moving existing /c/projects/cool-project/scripts to /c/projects/cool-project/updates/backups/19-08-21
mv: cannot move '/c/projects/cool-project/scripts' to '/c/projects/cool-project/updates/backups/19-08-21/scripts': Permission denied
Running command: mv '/c/projects/cool-project/updates/extracted/cool-project/scripts' '/c/projects/cool-project/scripts'
mv: cannot move '/c/projects/cool-project/updates/extracted/cool-project/scripts' to '/c/projects/cool-project/scripts/scripts': Directory not empty
```

Windows file locks suck since ever!

**Comment by hohwille** [comment]
I tried this in `doReplaceExtractedFile`:
```
  if [ "${target}" = "${DEVON_IDE_HOME}/scripts" ] && doIsWindows
  then
    echo "icacls $(cygpath -w "${target}") /reset /T"
    icacls "$(cygpath -w "${target}")" "/remove:d" '*S-1-1-0' "//T" "//C"
    icacls "$(cygpath -w "${target}")" "//reset" "//T"
  fi
```

However it is not working. I am not a windows expert and am not entirely sure if `icacls` can actually solve my problem...
https://ss64.com/nt/icacls.html

**Comment by hohwille** [comment]
This `devon-ide` really makes a hacker out of me (usually I am a programmer or coder who likes clean and simple solutions). However, trying all I can to solve this I came up with the following hack that seemd to work:
```
(sleep 10;mv "scripts" "update/backup";echo "moved scripts") &
```
It is just a PoC but with this crazy hack I think it will be possible to solve the issue.

**Comment by hohwille** [comment]
To get this timing of "sleep" and the further processing of the devon command somewhat in sync, we need to separate the update of the ide scripts from the rest. To avoid breaking the current devon API and making things much more complicated, I would suggest to introduce some tweak that detects the situation in the main flow of the program and aborts the further updates of tools. By printing some explanation to repeat the update command after that we should have a working solution somehow even for windows.

**Comment by markusschuh** [comment]
Replacing the "scripts" directory - while a "devon ide update" bash script runs that call several functions from scriptlets in that directory itself - does not sound as the most stable approach anyway for me - even when tests under macOS / linux may not have revealed issues.

My own approach would have been to have some "core" tool for update which has no dependencies inside the updated area.

**Comment by hohwille** [comment]
Fixed with PR #223.

**Comment by hohwille** [comment]
@markusschuh we also have to think about the API.
`devon ide update` is the single command that all the users of a project need to call if something has been fixed to get up-to-date. No matter if it is an update of the `devon-ide-scripts` package, the settings git repository, a tool version (e.g. updated via settings) that changed, or all of these together. Now the letter requires to rerun `devon ide update` on windows but not on other platforms that otherwise would have drawbacks just because of windows flaws.

But feel free to create a new issue suggesting that we should have two separate commands:
* one for update of the `devon-ide-scripts` themselves
* one for the update of settings and tools.

What would be good new names for the CLI if we would go for this change?
